ql/src/java/org/apache/hadoop/hive/ql/parse/repl/metric/MetricCollector.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.ql.parse.repl.metric;
+
+import org.apache.hadoop.hive.ql.parse.SemanticException;
+import org.apache.hadoop.hive.ql.parse.repl.metric.event.ReplicationMetric;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * MetricCollector.
+ * In memory collection of metrics
+ */
+public class MetricCollector {
+  private Logger LOG = LoggerFactory.getLogger(MetricCollector.class);
+  private Map<Long, ReplicationMetric> metricMap = new ConcurrentHashMap<>();
+  private final long maxSize;
+
+  private static volatile MetricCollector instance;
+
+  private MetricCollector(long maxSize){
+    this.maxSize = maxSize;
+  }
+
+  public static MetricCollector getInstance(long maxSize) {
```

Comment:
This interface seems to have problem. If it is singleton, then there is no meaning of maxSize for subsequent calls, as you will always select the precious instance.

Reply:
How do you pass maxSize then? Constructor is not exposed that you pass it just once

Reply:
I can add a init method but the problem with that is it can be overwritten and we don't want that

Reply:
when it is first time you call this method with getInstance(10) it gets initialized with 10. On subsequent calls caller calls getInstance(20), getInstance(100) etc, but it will get the same instance. This will mislead the caller.